scripts/build-layout-scales.js

```diff
@@ -0,0 +1,1271 @@
+          margin-top: ${value(scale)} !important;
+          margin-bottom: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--m-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .my${scale}-mm {
+          margin-top: ${value(scale)} !important;
+          margin-bottom: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--l-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .my${scale}-ml {
+          margin-top: ${value(scale)} !important;
+          margin-bottom: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--xl-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .my${scale}-mxl {
+          margin-top: ${value(scale)} !important;
+          margin-bottom: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`
+    \n/**
+      * Apply margin on the left and right.
+      *
+      * @group
+      * @example
+      * <div class='mx24 bg-darken10'>mx24</div>
+      * @memberof Margins
+      */`
+    );
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .mx${scale} {
+          margin-left: ${value(scale)} !important;
+          margin-right: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--m-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .mx${scale}-mm {
+          margin-left: ${value(scale)} !important;
+          margin-right: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--l-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .mx${scale}-ml {
+          margin-left: ${value(scale)} !important;
+          margin-right: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--xl-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .mx${scale}-mxl {
+          margin-left: ${value(scale)} !important;
+          margin-right: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    return css;
+  };
+
+  variantGenerators.singleMargin = function (scales) {
+    function value(v) {
+      return typeof v === 'string' ? `-${v[v.length - 1]}px` : `${v}px`;
```

Comment:
This seems wrong to me ... If the type is a string, e.g. `-full`, you don't want to prepend a hyphen or append a `px` unit, right?